admin/src/react-components/service-editor.js

```diff
@@ -377,23 +410,29 @@ class ConfigurationEditor extends Component {
         </CardContent>
         <Snackbar
           anchorOrigin={{ horizontal: "center", vertical: "bottom" }}
-          open={this.state.saved || !!this.state.saveError}
+          open={this.state.saved || !!this.state.saveError || !!this.state.warningMessage}
           autoHideDuration={10000}
-          onClose={() => this.setState({ saved: false, saveError: null })}
+          onClose={() => {
+            this.setState({ saved: false, saveError: null, warningMessage: null });
+          }}
         >
           <SnackbarContent
             className={clsx({
-              [this.props.classes.success]: !this.state.saveError,
-              [this.props.classes.warning]: !!this.state.saveError
+              [this.props.classes.success]: !this.state.saveError && !this.state.warningMessage,
+              [this.props.classes.warning]: !!this.state.saveError || !!this.state.warningMessage
             })}
             message={
               <span id="import-snackbar" className={this.props.classes.message}>
                 <Icon className={clsx(this.props.classes.icon, this.props.classes.iconVariant)} />
-                {this.state.saveError || "Settings saved."}
+                {this.state.saveError || this.state.warningMessage || (this.state.saved && "Settings saved.") || ""}
```

Comment:
Does it need the extra `|| ""`? Does this stay popped up every time you type?

Reply:
After I clear the `warningMessage`, there's a brief animation that clears the toast. But then as that thing is rendering, it changes color (because `warningMessage` is no longer set). I don't care so much that the color changes -- but I don't want people to accidentally read "settings saved" when they have not been saved. This can probably be handled more cleanly but I wanted to rewrite as little as possible about how this worked.